shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -172,9 +174,26 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContext(
   // the global handle at the right time.
   if (value->IsFunction()) {
     auto func = v8::Local<v8::Function>::Cast(value);
+    v8::MaybeLocal<v8::Value> maybe_original_fn = GetPrivate(
+        source_context, func, context_bridge::kOriginalFunctionPrivateKey);
 
     {
       v8::Context::Scope destination_scope(destination_context);
+      v8::Local<v8::Value> proxy_func;
+
+      // If this function has already been sent over the bridge,
+      // then it is being sent _back_ over the bridge and we can
+      // simply return the original method here for performance reasons
+
+      // For safety reasons we check if the destination context is the
+      // creation context of the original method.  If it's not we proceed
+      // with the proxy logic
+      if (maybe_original_fn.ToLocal(&proxy_func) && proxy_func->IsFunction() &&
+          proxy_func.As<v8::Object>()->CreationContext() ==
+              destination_context) {
```

Comment:
feels like some of these should be `CHECK` or maybe even `DCHECK`
```suggestion
      if (maybe_original_fn.ToLocal(&proxy_func)) {
        DCHECK(proxy_func->IsFunction());
        DCHECK(proxy_func.As<v8::Object>()->CreationContext() ==
               destination_context);
```

since i don't think there's any way these could _not_ be true?

Reply:
I can make the `CreationContext` check fail by using native modules are even just the `vm` module in NodeJS.  Because it was possible for legit user code to trip it I made it a silent condition rather than a hard `CHECK`